src/main/config/config_eeprom.c

```diff
@@ -83,6 +83,157 @@ typedef struct {
     uint32_t word;
 } PG_PACKED packingTest_t;
 
+#ifdef EEPROM_IN_EXTERNAL_FLASH
+bool loadEEPROMFromExternalFlash(void)
+{
+    const flashPartition_t *flashPartition = flashPartitionFindByType(FLASH_PARTITION_TYPE_CONFIG);
+    const flashGeometry_t *flashGeometry = flashGetGeometry();
+
+    uint32_t flashStartAddress = flashPartition->startSector * flashGeometry->sectorSize;
+
+    uint32_t totalBytesRead = 0;
+    int bytesRead = 0;
+
+    bool success = false;
+
+    do {
+        bytesRead = flashReadBytes(flashStartAddress + totalBytesRead, &eepromData[totalBytesRead], EEPROM_SIZE - totalBytesRead);
+        if (bytesRead > 0) {
+            totalBytesRead += bytesRead;
+            success = (totalBytesRead == EEPROM_SIZE);
+        }
+    } while (!success && bytesRead > 0);
+
+    return success;
+}
+#endif
+
+#ifdef EEPROM_IN_SDCARD
+
+enum {
+    FILE_STATE_NONE = 0,
+    FILE_STATE_BUSY = 1,
+    FILE_STATE_FAILED,
+    FILE_STATE_COMPLETE,
+};
+
+uint8_t fileState = FILE_STATE_NONE;
+
+const char *defaultSDCardConfigFilename = "CFCONFIG.BIN";
```

Comment:
`CFCONFIG.BIN` is not really an appropriate name for Betaflight. `CONFIG.BIN`?

Reply:
done